course_catalog/serializers.py

```diff
@@ -10,10 +10,30 @@
     CoursePrice,
     CourseTopic,
     Bootcamp,
+    LearningPathItem,
+    LearningPath,
+    Program,
+    ProgramItem,
 )
 from course_catalog.utils import get_ocw_topic, get_year_and_semester, get_course_url
 
 
+class GenericForeignKeyFieldSerializer(serializers.ModelSerializer):
+    """
+    Special field to handle the generic foreign key in ListItem
+    """
+
+    def to_representation(self, instance):
+        if isinstance(instance, Bootcamp):
+            serializer = BootcampSerializer(instance)
+        elif isinstance(instance, Course):
+            serializer = CourseSerializer(instance)
+        else:
+            raise Exception("Unexpected type of tagged object")
```

Comment:
Can you add a unit test for this condition? I think that may bring up the test coverage to ~95%.

Reply:
Added